test/levenberg_marquardt.jl

```diff
@@ -0,0 +1,97 @@
+let
+    function f_lm(x)
+      [x[1], 2.0 - x[2]]
+    end
+    function g_lm(x)
+      [1.0 0.0; 0.0 -1.0]
+    end
+
+    initial_x = [100.0, 100.0]
+
+    results = Optim.levenberg_marquardt(f_lm, g_lm, initial_x)
```

Comment:
Here and everywhere else this is called below, it should be `LsqFit.levenberg_marquardt`.